presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -613,4 +614,53 @@ public void testGreatCircleDistance()
         assertInvalidFunction("great_circle_distance(36.12, -86.67, 33.94, nan())", "Longitude must be between -180 and 180");
         assertInvalidFunction("great_circle_distance(36.12, -86.67, 33.94, infinity())", "Longitude must be between -180 and 180");
     }
+
+    @Test
+    public void testSTNumGeometries()
+    {
+        assertSTNumGeometries("MULTIPOINT EMPTY", 0);
+        assertSTNumGeometries("MULTILINESTRING EMPTY", 0);
+        assertSTNumGeometries("MULTIPOLYGON EMPTY", 0);
+        assertSTNumGeometries("GEOMETRYCOLLECTION EMPTY", 0);
+        assertSTNumGeometries("POINT (1 2)", 1);
+        assertSTNumGeometries("LINESTRING(77.29 29.07,77.42 29.26,77.27 29.31,77.29 29.07)", 1);
+        assertSTNumGeometries("POLYGON ((0 0, 0 1, 1 1, 1 0, 0 0))", 1);
+        assertSTNumGeometries("MULTIPOINT (1 2, 2 4, 3 6, 4 8)", 4);
+        assertSTNumGeometries("MULTILINESTRING ((1 1, 5 1), (2 4, 4 4))", 2);
+        assertSTNumGeometries("MULTIPOLYGON (((1 1, 1 3, 3 3, 3 1)), ((2 4, 2 6, 6 6, 6 4)))", 2);
+        assertSTNumGeometries("GEOMETRYCOLLECTION(POINT(2 3), LINESTRING (2 3, 3 4))", 2);
+    }
+
+    private void assertSTNumGeometries(String wkt, int expected)
+    {
+        assertFunction("ST_NumGeometries(ST_GeometryFromText('" + wkt + "'))", INTEGER, expected);
+    }
+
+    @Test
+    public void testSTGeometryN()
+    {
+        assertSTGeometryN("POINT (1 2)", 1, "POINT (1 2)");
```

Comment:
Would you group these tests by geometry type for readability? It seems that happy tests use only index == 1. Would you add some happy tests with index > 1?

Reply:
I submitted the change last Friday, while it show conflict. Any suggestion or hint?

Reply:
You need to rebase against the master (after you pull down the latest changes), and fix the merge conflicts.